src/libs/pluginprocess/pluginprocess.c

```diff
@@ -0,0 +1,379 @@
+}
+
+/** Call a plugin's function in a child process
+ *
+ * This will wrap all the required information to execute the given
+ * command in a keyset and send it over to the child process. Then
+ * it waits for the child process's answer and copies the result 
+ * back into the original plugin keyset and plugin key.
+ * 
+ * Typically called like
+ * @code
+int elektraPluginSet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessSend (pp, ELEKTRA_PLUGIN_SET, returned, parentKey);
+
+	// actual plugin functionality to be executed in a child process
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+ * @endcode
+ *
+ * @param pp the data structure containing the plugin's process information
+ * @param command the plugin command that should be executed, e.g. ELEKTRA_PLUGIN_GET
+ * @param originalKeySet the original key set that the parent process receives
+ * @param key the original key the parent process receives
+ * @retval ELEKTRA_PLUGIN_STATUS_ERROR if the child process communication failed
+ * @retval the called plugin's return value otherwise
+ * @see elektraPluginProcessIsParent for checking if we are in the parent or child process
+ * @ingroup processplugin
+ **/
+int elektraPluginProcessSend (const ElektraPluginProcess * pp, plugin_t command, KeySet * originalKeySet, Key * key)
+{
+	// Some plugin functions don't use keysets, but we need one for our
+	// simple communication protocol, so create a temporary one
+	KeySet * keySet = originalKeySet == NULL ? ksNew (3, KS_END) : ksDup (originalKeySet);
+	ksAppendKey (keySet, keyDup (key));
+	char * commandStr = intToStr (command);
+	ksAppendKey (keySet, keyNew ("/pluginprocess/command", KEY_VALUE, commandStr, KEY_END));
+	Key * originalKey = keyNew ("/pluginprocess/original", KEY_VALUE, keyName (key), KEY_END);
+	ksAppendKey (keySet, originalKey);
+
+	// Serialize, currently statically use dini our default format, this already writes everything out to the pipe
+	Key * commandPipeKey = keyNew ("/pluginprocess/commandPipe", KEY_VALUE, pp->commandPipe, KEY_END);
+	Key * resultPipeKey = keyNew ("/pluginprocess/resultPipe", KEY_VALUE, pp->resultPipe, KEY_END);
+	ELEKTRA_LOG ("P: We send %zd keys through the pipe now to issue command %d it", ksGetSize (keySet), command);
+	elektraInvoke2Args (pp->dump, "set", keySet, commandPipeKey);
+	ELEKTRA_LOG_DEBUG ("P: Waiting for the result now");
+	elektraInvoke2Args (pp->dump, "get", keySet, resultPipeKey);
+	ELEKTRA_LOG ("P: We received %zd keys in return", ksGetSize (keySet));
+
+	// Bring everything back in order by removing our process-related keys
+	Key * originalDeserializedKey = ksLookupByName (keySet, keyName (key), KDB_O_POP);
+	Key * resultKey = ksLookupByName (keySet, "/pluginprocess/result", KDB_O_POP);
+	char * endPtr;
+	int prevErrno = errno;
+	errno = 0;
+	int result = (int)strtol (keyString (resultKey), &endPtr, 10);
```

Comment:
This cast is ugly (why not store it in a long int?) and error checks are missing.

Reply:
the error checks are further down below, i guess this is not a good idea since you've overlooked them :) I'll refactor it.